File: webservice/src/main/java/com/paul/mcgranaghan/webserver/dto/NameBasics.java
package com.paul.mcgranaghan.webserver.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.util.Arrays;
import java.util.Set;
import java.util.stream.Collectors;

@Getter
@Builder(toBuilder = true)
@AllArgsConstructor
@NoArgsConstructor
public class NameBasics {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    public String _id;
    public String nConst;
    public String primaryName;
    public String birthYear;
    public String deathYear;
    public String primaryProfession;
    public String knownForTitles;
    public Set<Profession> primaryProfessions;
    public Set<String> knownTitles;

    public NameBasics(String _id, String nConst,
                      String primaryName, String birthYear,
                      String deathYear, String primaryProfession,
                      String knownForTitles) {
        this._id = _id;
        this.nConst = nConst;
        this.primaryName = primaryName;
        this.birthYear = birthYear;
        this.deathYear = deathYear;
        this.primaryProfession = primaryProfession;
        this.knownForTitles = knownForTitles;

        this.primaryProfessions = Arrays.stream(primaryProfession.split(","))
                .map(p -> Profession.valueOf(p.toUpperCase())).collect(Collectors.toSet());
        this.knownTitles = Arrays.stream(knownForTitles.split(",")).collect(Collectors.toSet());
    }

    public void setPrimaryProfessionsFromString(String professions) {
        this.primaryProfessions = Arrays.stream(primaryProfession.split(","))
                .map(p -> Profession.valueOf(p.toUpperCase())).collect(Collectors.toSet());

    }
}

File: filesync/dto/NameBasics.py
import pandas
from dataclasses import dataclass

from filesync.dto.Profession import Profession


@dataclass
class NameBasics:
    def __init__(self, _id, n_const, primary_name, birth_year, death_year,
                 primary_profession, known_for_titles):
        self._id: str = _id
        self.n_const: str = n_const
        self.primary_name: str = primary_name
        self.birth_year: str = birth_year
        self.death_year: str = death_year
        self.primary_profession: set[Profession] = primary_profession
        self.known_for_titles: set[str] = known_for_titles


def df_to_list(df: pandas.DataFrame) -> list:
    return list(map(lambda x: NameBasics(_id=x[0], n_const=x[1], primary_name=x[2],
                                         birth_year=x[3], death_year=x[4],
                                         primary_profession=x[5],
                                         known_for_titles=x[6]), df.values.tolist()))
